specificatie/zoek-personen.yaml

```diff
@@ -0,0 +1,324 @@
+              pattern: ^[a-zA-Z0-9À-ÿ' '\-\']{1,10}$
+              example: "de"
+            voornamen:
+              description: |
+                De verzameling namen die, gescheiden door spaties, aan de geslachtsnaam voorafgaat. ** Bij deze query-parameter is het gebruik van een [wildcard](https://github.com/VNG-Realisatie/Haal-Centraal-common/blob/v1.3.0/features/wildcard.feature) toegestaan in combinatie met minimaal 2 karakters.** **Zoeken met tekstvelden is [case-Insensitive](https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/blob/v1.3.0/features/case_insensitive.feature).**
+              type: string
+              pattern: ^(\?+|\*)?[a-zA-Z0-9À-ÿ' '\-\']{1,200}(\?+|\*)?$
+              example: "Dirk"
+    ZoekMetPostcodeEnHuisnummer:
+      required:
+        - postcode
+        - huisnummer
+      allOf:
+        - $ref: '#/components/schemas/PersonenQuery'
+        - type: object
+          properties:
+            inclusiefOverledenPersonen:
+              $ref: '#/components/schemas/InclusiefOverledenPersonen'
+            huisletter:
+              $ref: 'verblijfplaats.yaml#/components/schemas/Huisletter'
+            huisnummer:
+              $ref: 'verblijfplaats.yaml#/components/schemas/Huisnummer'
+            huisnummertoevoeging:
+              $ref: 'verblijfplaats.yaml#/components/schemas/Huisnummertoevoeging'
+            postcode:
+              $ref: 'verblijfplaats.yaml#/components/schemas/Postcode'
+    ZoekMetStraatHuisnummerEnGemeenteVanInschrijving:
+      required:
+        - straat
+        - huisnummer
+        - gemeenteVanInschrijving
+      allOf:
+        - $ref: '#/components/schemas/PersonenQuery'
+        - type: object
+          properties:
+            inclusiefOverledenPersonen:
+              $ref: '#/components/schemas/InclusiefOverledenPersonen'
+            huisletter:
+              $ref: 'verblijfplaats.yaml#/components/schemas/Huisletter'
+            huisnummer:
+              $ref: 'verblijfplaats.yaml#/components/schemas/Huisnummer'
+            huisnummertoevoeging:
+              $ref: 'verblijfplaats.yaml#/components/schemas/Huisnummertoevoeging'
+            straat:
+              $ref: 'filter.yaml#/components/schemas/StraatFilter'
+    ZoekMetNummeraanduidingIdentificatie:
+      required:
+        - nummeraanduidingIdentificatie
+      allOf:
+        - $ref: '#/components/schemas/PersonenQuery'
+        - type: object
+          properties:
+            inclusiefOverledenPersonen:
+              $ref: '#/components/schemas/InclusiefOverledenPersonen'
+            nummeraanduidingIdentificatie:
+              $ref: 'verblijfplaats.yaml#/components/schemas/NummeraanduidingIdentificatie'
+    PersonenQueryResponse:
+      type: object
+      required:
+        - type
+      discriminator:
+        propertyName: type
+        mapping:
+          ZoekMetGeslachtsnaamEnGeboortedatum: '#/components/schemas/ZoekMetGeslachtsnaamEnGeboortedatumResponse'
```

Comment:
is er een reden om een response te definiëren voor elk zoekpad? Hadden dat er niet twee kunnen zijn: een voor zoeken en een voor raadplegen?

Reply:
Ja, dit is ten behoeve van serialization en onze keuze om voor de mapping namen paren (dus bijv. ZoekMetPostcodeEnHuisnummer met bijbehorende ZoekMetPostcodeEnHuisnummerResponse) te gebruiken. Als je maar één response type hebt voor zoeken, dan kan de serializer niet bepalen welke mapping naam moet worden gebruikt. Dit kan alleen worden opgelost door de mapping namen paren constraint los te laten

Reply:
ik denk dat wanneer je de response ook zou definiëren in zowel een GBA variant als een Haal Centraal variant, de berichtdefinities voor zowel de GBA versie als m.n. voor de gebruikers achter de proxy eenvoudiger zou worden en die laatste geen GBA componenten meer hoeft te hebben. Bijvoorbeeld geheimhouding en inOnderzoek kan je dan overal direct in de juiste vorm specificeren en ook abstract datums hebben dan geen GBA versie meer

Reply:
done

Reply:
> Ja, dit is ten behoeve van serialization en onze keuze om voor de mapping namen paren (dus bijv. ZoekMetPostcodeEnHuisnummer met bijbehorende ZoekMetPostcodeEnHuisnummerResponse) te gebruiken.

Dit soort keuzes documenteren we volgens mij nu niet. We zijn ooit eens begonnen met het vastleggen van Design Decisions. Is het een idee om dit soort belangrijke keuzes te documenteren ?

Reply:
Overigens wil ik best alle discussie en commentaren doorlopen om te kijken welke design decisions ik boven water kan krijgen.